src/coordinator.coffee

```diff
@@ -97,9 +97,10 @@ pingUrl = (address, method, callback) ->
 
 class Coordinator extends EventEmitter
   constructor: (@broker, @options = {}) ->
-    @participants = {}
+    @participants = {} # participantId -> Definition (from discovery)
     @connections = {} # connId -> { queue: opt String, handler: opt function }
     @iips = {} # iipId -> { metadata, data }
+    @nodes = {} # role -> { metadata: {} }
```

Comment:
Yes that was my intention. Question remains what to do with discovered things which are not part of `nodes`. They have to be in the graph somehow, otherwise Flowhub will never know about them and it wont be possible to add them?